vsutil/types.py

```diff
@@ -5,12 +5,18 @@
 
 # this file only depends on the stdlib and should stay that way
 from enum import Enum
-from typing import Any, Optional, Type, TypeVar, Union
+from typing import Any, Callable, Optional, Type, TypeVar, Union
 
 E = TypeVar('E', bound=Enum)
 
 
-class Dither(str, Enum):
+class _NoSubmoduleRepr(Enum):
```

Comment:
I would define `_NoSubmoduleRepr` as a mixin instead of subclassing Enum with it, i.e. `class Range(int, _NoSubmoduleRepr, Enum):`.

Reply:
Good idea, changed in https://github.com/Irrational-Encoding-Wizardry/vsutil/pull/50/commits/80a21103365fd9f0e303ac744a075a71ee65be6f with a slight difference. The mixin class MUST come first otherwise `int` and `str`'s `__repr__` overrides our custom one, you can see that here: https://nbviewer.jupyter.org/gist/OrangeChannel/49a58bb6af63fc1baddd30ed68574479